lib/fastlane/plugin/android_emulator/actions/android_emulator_action.rb

```diff
@@ -40,9 +49,14 @@ def self.run(params)
 
         UI.message("Starting emulator")
         system("LC_NUMERIC=C; #{sdk_dir}/tools/emulator @#{params[:name]} > /dev/null 2>&1 &")
-        sh("#{adb} -e wait-for-device")
 
-        until Actions.sh("#{adb} -e shell getprop init.svc.bootanim", log: false).include? "stopped" do
+        error_callback = proc do |error|
+          UI.error("Emulator may be having issues... or could have started too quickly!")
+        end
+        Actions.sh("#{adb} -e wait-for-device", error_callback: error_callback)
```

Comment:
Yeah, so without the sleep, many times it failed.
After adding 2s sleep it failed also a couple of times.

What do you think, if we try this command a couple of times and catch the errors?

Reply:
That would be good -- and also to timeout the `wait-for-device` call after a (user configurable?) time. I've noticed that if the emulator fails to start completely, then the build can hang at this point indefinitely.

I'm not sure how to timeout a shell call within fastlane... The Actions.sh function doesn't seem to support it.